packages/application-generic/src/usecases/process-tenant/process-tenant.usecase.ts

```diff
@@ -0,0 +1,122 @@
+import { Injectable } from '@nestjs/common';
+import { TenantRepository, TenantEntity } from '@novu/dal';
+import { ITenantDefine } from '@novu/shared';
+import { isEqual } from 'lodash';
+
+import { InstrumentUsecase } from '../../instrumentation';
+import { ProcessTenantCommand } from './process-tenant.command';
+import { UpdateTenant, UpdateTenantCommand } from '../update-tenant';
+import { CreateTenant, CreateTenantCommand } from '../create-tenant';
+
+@Injectable()
+export class ProcessTenant {
+  constructor(
+    private updateTenantUsecase: UpdateTenant,
+    private createTenantUsecase: CreateTenant,
+    private tenantRepository: TenantRepository
+  ) {}
+
+  @InstrumentUsecase()
+  public async execute(
+    command: ProcessTenantCommand
+  ): Promise<TenantEntity | undefined> {
+    const { environmentId, organizationId, userId, tenant } = command;
+
+    let tenantEntity;
+
+    try {
+      tenantEntity = await this.getTenant(
+        environmentId,
+        organizationId,
+        userId,
+        tenant
+      );
+    } catch (e) {
+      tenantEntity = null;
```

Comment:
Why do we do null here? will getTenant in some cases return null otherwise we could reduce this code with just do the catch with no code and skip the null check on line 38?